File: solutions.digamma.damas.jcr/src/main/java/solutions/digamma/damas/jcr/content/JcrDetailedDocument.java
package solutions.digamma.damas.jcr.content;

import solutions.digamma.damas.common.WorkspaceException;
import solutions.digamma.damas.content.DetailedDocument;
import solutions.digamma.damas.content.Version;
import solutions.digamma.damas.inspection.NotNull;

import javax.jcr.Node;
import java.util.ArrayList;
import java.util.List;

/**
 * @author Ahmad Shahwan
 */
public class JcrDetailedDocument extends JcrDocument
        implements DetailedDocument, JcrDetailedFile, JcrCommentReceiver{

    /**
     * Constructor with JCR node.
     *
     * @param node
     */
    JcrDetailedDocument(@NotNull Node node) throws WorkspaceException {
        super(node);
    }

    @Override
    public @NotNull List<@NotNull Version> getVersions() {
        return new ArrayList<>(0);
    }
}

File: solutions.digamma.damas.jcr/src/main/kotlin/solutions/digamma/damas/jcr/content/JcrDetailedDocument.kt
package solutions.digamma.damas.jcr.content

import solutions.digamma.damas.common.WorkspaceException
import solutions.digamma.damas.content.DetailedDocument
import solutions.digamma.damas.content.Version
import java.util.ArrayList
import javax.jcr.Node

/**
 * Detailed document.
 *
 * @author Ahmad Shahwan
 *
 * @param node
 */
class JcrDetailedDocument
@Throws(WorkspaceException::class)
private constructor(node: Node) : JcrDocument(node), DetailedDocument,
        JcrDetailedFile, JcrCommentReceiver {

    override fun getVersions(): List<Version> {
        return ArrayList(0)
    }

    companion object {

        @Throws(WorkspaceException::class)
        fun of(node: Node) = JcrDetailedDocument(node)
    }
}
